eos-tech-support/eos-hack-reset-quest

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+usage () {
+    cat <<EOF
+Usage: $0 QUEST-NAME|all
+Reset the state of QUEST-NAME, you can specify "all",
+to do a full reset of all the quests and a reset of the
+desktop icons generated by quests.
+
+  -h, --help                display this help and exit
+
+EOF
+}
+
+args=$(getopt -o h -l "help" -n "eos-hack-reset-quests" -- "$@")
+
+if [ $? != 0 ]; then
+    exit 1
+fi
+
+eval set -- "$args"
+
+quest_name=""
+while true; do
+    case "$1" in
+        -h|--help)
+        usage
+        exit 0
+        ;;
+    --)
+        shift
+        ;;
+    *)
+        quest_name=$1
+        break
+        ;;
+    esac
+done
+
+if [ -z $quest_name ]; then
+    echo "Please specify a quest name or 'all'"
+    exit 1
+elif [ $quest_name == 'all' ]; then
+    rm -rf ~/.var/app/com.endlessm.GameStateService/data/*
+    eos-reset-desktop
+else
+    state_filename=~/.var/app/com.endlessm.GameStateService/data/state.json
+
+    flatpak run --command=true com.endlessm.HackToolbox 2> /dev/null
+    if [ $? != 0 ]; then
+        echo "You have to install 'com.endlessm.HackToolbox' in order to reset a single quest"
+        exit 1
+    fi
+
+    flatpak run --command=jq --filesystem=~/.var/app/com.endlessm.GameStateService \
+        com.endlessm.HackToolbox \
```

Comment:
This is weird: run app A, giving it access to app B's state, and manually modify the config file. This ties the file format of GameStateService (maintained outside the OS) to this script (maintained in the OS) so you're committing to supporting this file format forever. I think it would be better to ship this tool as part of GameStateService and just have this OS-level script run `flatpak run --command=eos-hack-reset com.endlessm.GameStateService "$@"`.

Reply:
I just wanted to use `jq` :(